voyager-core/src/commonMain/kotlin/cafe/adriel/voyager/core/screen/Screen.kt

```diff
@@ -0,0 +1,14 @@
+package cafe.adriel.voyager.core.screen
+
+import androidx.compose.runtime.Composable
+
+public expect interface Screen {
+
+    public open val key: ScreenKey
+
+    @Composable
+    public fun Content()
+}
+
+internal fun Screen.commonKeyGeneration() =
+    this::class.qualifiedName ?: error("Default ScreenKey not found, please provide your own key")
```

Comment:
This was required because Kotlin currently does not support default implementation on expect classes. You can follow issue here https://youtrack.jetbrains.com/issue/KT-20427